### PR TITLE
Feature/test framework

### PR DESCRIPTION
**Pull request by khdegraaf**
Create a copy of cosmos-sdk SimApp for Sifchain, and redo CLP module to use it.  Should increase test coverage a bit, and allow full module integration testing.

### CODE REVIEW

File: x/clp/keeper/keeper.go [hunk 1]
```diff
@@ -12,7 +12,7 @@ import (
 // Keeper of the clp store
 type Keeper struct {
 	storeKey   sdk.StoreKey
-	cdc        *codec.Codec
+	Cdc        *codec.Codec
```
Comment (Jedi2002) [inline x/clp/keeper/keeper.go:15]: I think it would be a better Idea , to create getter for this isntead of exposing it directly

**Review by Jedi2002 — DISMISSED** [review]
Looks good , minor comment

File: x/clp/keeper/keeper.go (hunk 1 above)
Comment (khdegraaf) [reply x/clp/keeper/keeper.go:15]: Fair point, I'll fix this.

**Review by kingpinXD — DISMISSED** [review]
Lgtm

File: simapp/app_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	app.InitChain(
+		abci.RequestInitChain{
+			Validators:    []abci.ValidatorUpdate{},
+			AppStateBytes: stateBytes,
+		},
+	)
+	app.Commit()
+
+	// Making a new app object with the db, so that initchain hasn't been called
+	app2 := NewSimApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, map[int64]bool{}, 0)
```
Comment (utx0) [inline simapp/app_test.go:34]: whats the point with the second app object?

File: simapp/app_test.go [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	//_, _, err = app2.ExportAppStateAndValidators(false, []string{})
+	//require.NoError(t, err, "ExportAppStateAndValidators should not have an error")
+	require.NotNil(t, app2)
+}
+
+// ensure that black listed addresses are properly set in bank keeper
+/* func TestBlackListedAddrs(t *testing.T) {
```
Comment (utx0) [inline simapp/app_test.go:41]: What this all about?

Comment (utx0) [reply simapp/app_test.go:41]: maybe just some clean up?

**Review by utx0 — COMMENTED** [review]
Looks like this is coming alone nicely, however @khdegraaf I might need to do a pairing session on some of this to make sure I understand all the moving parts here correctly.

File: simapp/app_test.go (hunk 3 above)
Comment (khdegraaf) [reply simapp/app_test.go:41]: I took the cosmos-sdk version of simapp stuff, and trimmed it down and modified it to match our app.  Some of this I could just remove, but need to talk about whether it is stuff we will support long-term or not.  I know some of it is functionality we just don't support yet.  Like Slashing and Genesis.  Others may not be relevant.  Probably need to have a discussion on some of the details.

File: simapp/app_test.go (hunk 2 above)
Comment (khdegraaf) [reply simapp/app_test.go:34]: Make sure we can do this twice.  Fuzz testing will do this over and again multiple times.

**Comment by khdegraaf** [comment]
OK, I cleaned up things we discussed @utx0

**Review by utx0 — APPROVED** [review]
Nice one!

**Pull request merged**